File: frontend/src/types/booking.ts
export interface Booking {
  id: number;
  customerId: number;
  customerName: string;
  vehicleId: number;
  vehiclePlateNumber: string;
  vehicleMake: string;
  vehicleModel: string;
  serviceId: number;
  serviceName: string;
  bookingDate: string;
  bookingTime: string;
  status: string;
  isDeleted: boolean;
  notes: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateBookingRequest {
  vehicleId: number;
  serviceId: number;
  bookingDate: string;
  bookingTime: string;
  notes: string | null;
}

export interface UpdateBookingRequest {
  vehicleId: number;
  serviceId: number;
  bookingDate: string;
  bookingTime: string;
  notes: string | null;
}

export interface UpdateBookingStatusRequest {
  status: string;
}

export interface BookingAvailability {
  date: string;
  serviceId: number;
  availableTimes: string[];
}

export interface CreateBookingFormErrors {
  vehicleId?: string;
  serviceId?: string;
  bookingDate?: string;
  bookingTime?: string;
  notes?: string;
}

File: backend/PrimeKare.Api/Models/Booking.cs
namespace PrimeKare.Api.Models;

public class Booking
{
    public int Id { get; set; }

    public int CustomerId { get; set; }
    public Customer Customer { get; set; } = null!;

    public int VehicleId { get; set; }
    public Vehicle Vehicle { get; set; } = null!;

    public int ServiceId { get; set; }
    public Service Service { get; set; } = null!;

    public DateOnly BookingDate { get; set; }

    public TimeSpan BookingTime { get; set; }

    public string Status { get; set; } = "pending";

    public string? Notes { get; set; }

    public bool IsDeleted { get; set; } = false;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}